units/crons.go

```diff
@@ -553,11 +553,26 @@ func PopulateAgentDeployJobs(env evergreen.Environment) amboy.QueueOperation {
 			return nil
 		}
 
-		hosts, err := host.Find(host.NeedsNewAgent(time.Now()))
+		hosts, err := host.Find(host.LastCommunicationTimeElapsed(time.Now()))
 		grip.Error(message.WrapError(err, message.Fields{
 			"operation": "background task creation",
 			"cron":      agentDeployJobName,
 			"impact":    "agents cannot start",
+			"message":   "problem finding hosts with elapsed last communication time",
+		}))
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		for _, h := range hosts {
+			h.SetNeedsNewAgent(true)
+		}
```

Comment:
should this be a multi update? 

what happens if there are nothing to reset? should there be some early exit rather than running another query?

Reply:
also this seems like a lot to have happen in the crons, which means that it's going to get set/run on every app server, which seems a bit extreme.